examples/lump-mpi.py

```diff
@@ -0,0 +1,171 @@
+"""
+import logging
+import numpy as np
+import pyopencl as cl
+import numpy.linalg as la  # noqa
+import pyopencl.array as cla  # noqa
+from functools import partial
+from mpi4py import MPI
+
+from meshmode.array_context import PyOpenCLArrayContext
+from meshmode.dof_array import thaw
+from meshmode.mesh import BTAG_ALL, BTAG_NONE  # noqa
+from grudge.eager import EagerDGDiscretization
+from grudge.shortcuts import make_visualizer
+
+
+from mirgecom.euler import inviscid_operator
+from mirgecom.simutil import (
+    inviscid_sim_timestep,
+    exact_sim_checkpoint
+)
+from mirgecom.io import make_init_message
+
+from mirgecom.integrators import rk4_step
+from mirgecom.steppers import advance_state
+from mirgecom.boundary import PrescribedBoundary
+from mirgecom.initializers import Lump
+from mirgecom.eos import IdealSingleGas
+
+
+def main(ctx_factory=cl.create_some_context):
+
+    cl_ctx = ctx_factory()
+    queue = cl.CommandQueue(cl_ctx)
+    actx = PyOpenCLArrayContext(queue)
+
+    logger = logging.getLogger(__name__)
+
+    dim = 3
+    nel_1d = 16
+    order = 3
+    exittol = .09
+    t_final = 0.01
+    current_cfl = 1.0
+    vel = np.zeros(shape=(dim,))
+    orig = np.zeros(shape=(dim,))
+    vel[:dim] = 1.0
+    current_dt = .001
+    current_t = 0
+    eos = IdealSingleGas()
+    initializer = Lump(numdim=dim, center=orig, velocity=vel)
+    casename = 'lump'
+    boundaries = {BTAG_ALL: PrescribedBoundary(initializer)}
+    constant_cfl = False
+    nstatus = 1
+    nviz = 1
+    rank = 0
+    checkpoint_t = current_t
+    current_step = 0
+    timestepper = rk4_step
+    box_ll = -5.0
+    box_ur = 5.0
+
+    comm = MPI.COMM_WORLD
+    nproc = comm.Get_size()
+    rank = comm.Get_rank()
+    num_parts = nproc
+
+    from meshmode.distributed import (
+        MPIMeshDistributor,
+        get_partition_by_pymetis,
+    )
+
+    mesh_dist = MPIMeshDistributor(comm)
+    global_nelements = 0
+    local_nelements = 0
+    if mesh_dist.is_mananger_rank():
+        from meshmode.mesh.generation import generate_regular_rect_mesh
+
+        mesh = generate_regular_rect_mesh(
+            a=(box_ll,) * dim, b=(box_ur,) * dim, n=(nel_1d,) * dim
+        )
+        global_nelements = mesh.nelements
+        logging.info(f"Total {dim}d elements: {global_nelements}")
+
+        part_per_element = get_partition_by_pymetis(mesh, num_parts)
+
+        local_mesh = mesh_dist.send_mesh_parts(mesh, part_per_element, num_parts)
+        del mesh
+
+    else:
+        local_mesh = mesh_dist.receive_mesh_part()
+    local_nelements = local_mesh.nelements
+
+    discr = EagerDGDiscretization(
+        actx, local_mesh, order=order, mpi_communicator=comm
+    )
+    nodes = thaw(actx, discr.nodes())
+    current_state = initializer(0, nodes)
+
+    visualizer = make_visualizer(discr, discr.order + 3
+                                 if discr.dim == 2 else discr.order)
+    initname = initializer.__class__.__name__
+    eosname = eos.__class__.__name__
```

Comment:
Maybe, but I don't *always* want to name the case and init just according to the class names. I am just doing it here because I rolled the examples that differ only according to their initializer and eos.